File: clients/typescript/src/bots/myAI.ts
import { BotAI } from './botAI';
import { PlayState } from './data';

export class MyAI implements BotAI {
  private name: string = 'YourBotName';

  play(currentGameState: PlayState): number {
    console.log("Your Coin ID is: " + currentGameState.coin_id);
    return 0;
  }

  getName(): string {
    return this.name;
  }
}

File: clients/csharp/Bots/MyAi.cs
using CsClient.Bots.Internal;

namespace CsClient.Bots
{
    public class MyAi : IBot
    {
        private int MyCoinID = -1;

        /// <inheritdoc/>
        public string Name { get; }

        /// <inheritdoc/>
        public int Play(PlayState? playState)
        {
            if (MyCoinID == -1)
            {
                MyCoinID = playState!.CoinId;
            }

            // put your logic here
            return 0;
        }

        public MyAi()
        {
            // give your bot a super duper cool name
            Name = "YourNameHere";
        }
    }
}